### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by coderschoolreview** [thanhcs94/CS_MLCourse_Week3#1]
The goal of this assignment was to introduce you to three new Natural Language Processing techniques, and to understand how to perform some basic sentiment analysis on song lyrics using these methods. You learned how to clean and prepare textual information for NLP, and then apply the following approaches: Bag Of Words, TF-IDF, and Doc2Vec. You used your prior knowledge of Python estimators, feature selection, and parameter optimization techniques to produce feature vectors from these NLP methods to make predictions on the moods of songs using their lyrics.

Unfortunately you have made a simple mistake which completely makes all your results wrong. I am also curious how come your scores seem so high -- when I re-ran your notebook, I got extremely low scores. I will message you on gitter about this.

The mistake you made is the following:

When creating your `train_test_split`, `X` needs to be your features and `y` needs to be your labels. So in this assignment, your features are your bag of words / tf-idf vectors, and your labels are your moods. However, you set your labels to be the lyrics...

What you did: 
`X = bag_of_words`
`y = df_songs_final['lyrics_features_clean']`

But what it should have been: 
`X = bag_of_words`
`y = df_songs_final['mood_filter']`

Also, you did not reset the index on your final dataframe, `df_songs_final`. Instead you did it very early (on `df_songs_lyrics`). Because of this, your song indexes did not map to the `Doc2Vec` feature vector indexes in the `.txt` file.

Let us chat about this on gitter soon so we can clear up any confusion you may have.
